src/v2/middleware/authenticate.go

```diff
@@ -136,21 +144,17 @@ func decodeApiKey(headerValue string, w http.ResponseWriter) apikey.Claims {
 }
 
 func performAPIKeyAuthentication(claims apikey.Claims, w http.ResponseWriter, r *http.Request) {
-	// TODO: Update this with v2 logic for Api keys
-	u, err := v1Handlers.GetUser(claims.UserID)
-	if err != nil {
-		m := "Invalid API Key, Authorization failed"
-		error_handler.Exit(http.StatusUnauthorized, m)
-	}
+	individualId := r.Header.Get(config.IndividualHeaderKey)
 
 	t := token.AccessToken{}
-	t.IamID = u.IamID
-	t.Name = u.Name
-	t.Email = u.Email
-
 	token.Set(r, t)
-	token.SetUserID(r, u.ID.Hex())
-	token.SetUserRoles(r, v1Handlers.GetUserRoles(u.Roles))
+	if len(strings.TrimSpace(individualId)) != 0 {
+		token.SetUserID(r, individualId)
+		token.SetUserRole(r, rbac.ROLE_USER)
+	}
+	if claims.Scopes == "config" || claims.Scopes == "onboard" || claims.Scopes == "audit" {
+		token.SetUserRole(r, rbac.ROLE_ADMIN)
```

Comment:
User role should not be set based on scope, rather based on presence of individualId in header.